lib/anyway/config.rb

```diff
@@ -135,6 +158,18 @@ def load_from_env(config)
       config
     end
 
+    def option_parser
+      build_option_parser if @option_parser.nil?
+      @option_parser
+    end
+
+    def parse_options!(options)
+      option_parser.parse!(options)
+    rescue OptionParser::InvalidOption
+      # NOTE: Do not fail when unknown arguments were given
```

Comment:
Why do we want to ignore unknown arguments?

Reply:
When unknown arguments are passes to `parse_options!` method, like:

```ruby
class MyConfig << Anyway::Config
  attr_config :host, :port
end

...

config.parse_option!(%w[--host localhost --port 3000 --log-level debug])
```

the mehtod call will raise an `OptionParser::InvalidOption` exception with the message "unknown argument '--log-level'"`. I don't think it's a good behavior to raise an exception in this case.

Reply:
@jastkand  What about this one?

Reply:
> If a developers don't want to raise exceptions they could rescues them themselves

That's fair.

Anyway, let me confirm if I correctly understand the behavior which you expect. Let's say we have a class:

```ruby
class MyConfig < Anyway::Config
  attr_config :host, :port, :log_level

  ignore_options :log_level
end
```

Should it raise the `OptionParser::InvalidOption` exception when I call:

```ruby
config.parse_option!(%w[--host localhost --port 3000 --log-level debug])
```

Reply:
Yep, it should raise an exception